core/schema/src/main/java/org/hawkular/metrics/schema/SchemaService.java

```diff
@@ -39,67 +41,125 @@
 
     private static Logger logger = LoggerFactory.getLogger(SchemaService.class);
 
-    public void run(Session session, String keyspace, boolean resetDB) {
-        run(session, keyspace, resetDB, 1);
+    private PreparedStatement versionUpdateQuery;
+
+    private Session session;
+
+    private String keyspace;
+
+    private Cassalog cassalog;
+
+    public SchemaService(Session session, String keyspace) {
+        this.keyspace = keyspace;
+        this.session = session;
     }
 
-    public void run(Session session, String keyspace, boolean resetDB, int replicationFactor) {
-        run(session, keyspace, resetDB, replicationFactor, false);
+    public void run(boolean resetDB) {
+        run(resetDB, 1);
     }
 
-    public void run(Session session, String keyspace, boolean resetDB, int replicationFactor, boolean updateVersion) {
+    public void run(boolean resetDB, int replicationFactor) {
+        run(resetDB, replicationFactor, false);
+    }
+
+    public void run(boolean resetDB, int replicationFactor, boolean updateVersion) {
+
+        URI script = getScript();
+        Map<String, ?> vars = this.getVars(resetDB, replicationFactor, true);
+        buildCassalogIfNeeded();
+        cassalog.execute(script, this.getTags(), vars);
+
+        if (updateVersion) {
+            updateVersion(0, 0);
+        }
+    }
+
+    private PreparedStatement prepareUpdateStatementIfNeeded() {
+        if(this.versionUpdateQuery == null) {
+            versionUpdateQuery = session.prepare("INSERT INTO sys_config (config_id, name, value) VALUES (?, ?, ?)");
+        }
+        return this.versionUpdateQuery;
+    }
+
+    private void buildCassalogIfNeeded() {
         CassalogBuilder builder = new CassalogBuilder();
-        Cassalog cassalog = builder.withKeyspace(keyspace).withSession(session).build();
-        Map<String, ?> vars  = ImmutableMap.of(
-                "keyspace", keyspace,
-                "reset", resetDB,
-                "session", session,
-                "replicationFactor", replicationFactor,
-                "logger", logger
-        );
+        cassalog = builder.withKeyspace(keyspace).withSession(session).build();
+    }
+
+    private Map<String, ?> getVars(boolean resetDB, int replicationFactor, boolean createSchema) {
```

Comment:
I don't think we want the ```createSchema``` variable. It is hard coded to true, so I don't see any need for it. It could potentially break the schema installs/upgrade which I will describe below.